### DISCUSSION TITLE
M2 Mac unknown architecture error

### DISCUSSION THREAD

**Discussion by joncokisler** [evilmartians/lefthook#390] [Q&A] [answered]
Hi,
I am trying to install lefthook again, since it stopped working suddenly on my M2 Macbook Pro.  I am getting this error: 
```
jon@Jons-MacBook-Pro ~ % lefthook install -f
Traceback (most recent call last):
	2: from /Users/jon/.rbenv/versions/2.6.6/bin/lefthook:23:in `<main>'
	1: from /Users/jon/.rbenv/versions/2.6.6/bin/lefthook:23:in `load'
/Users/jon/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/lefthook-1.2.2/bin/lefthook:12:in `<top (required)>': Unknown architecture:  (RuntimeError)
```
And when I  check what architecture leefthook recognizes my mac as:
```
irb(main):001:0> platform = Gem::Platform.new(RUBY_PLATFORM)
=> #<Gem::Platform:0x000000014d33b8c0 @cpu="", @os="darwin", @version="21" @cpu="", @os="darwin", @version="21">
```
As you can see, the cpu section is empty, that is why I am getting the `Unknown architecture` error. Maybe this is specific to M2 macs? I need some help :()

<img width="206" alt="Screen Shot 2022-11-23 at 11 48 29 AM" src="https://user-images.githubusercontent.com/12765701/203602742-c605682b-9122-467e-9090-06aeb8a8938f.png">

**Comment by mrexox** [comment]
Hey! Thank you for the issue! I guess there is something wrong with the ruby interpreter 🤔 but I need to ask my colleagues who has M2 to make sure. Maybe it can be fixed for the gem.

I recommend installing with either  
`go install github.com/evilmartians/lefthook@latest` or with `brew install lefthook`. You can also download the binary from release attachments.

**Comment by mrexox** [comment, marked as answer]
Hey! It doesn't look like an error on all M2 chips, because I asked a few people who have them, and they don't have such an issue. I think there was a problem with compilation ruby. Could you try rebuilding it and check it?

**Reply by joncokisler** [reply]
Thanks for looking into it! Do you mean rebuild ruby and try to install lefthook?

**Reply by mrexox** [reply]
Yes, I hope this will help.
